gcloud/_helpers.py

```diff
@@ -302,14 +303,30 @@ def _total_seconds(offset):
 def _rfc3339_to_datetime(dt_str):
     """Convert a string to a native timestamp.
 
+    .. note::
+
+       Python datetimes do not support nanosecond precision;  this function
+       therefore truncates such values to microseconds.
+
     :type dt_str: str
     :param dt_str: The string to convert.
 
     :rtype: :class:`datetime.datetime`
     :returns: The datetime object created from the string.
     """
-    return datetime.datetime.strptime(
-        dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    try:
+        return datetime.datetime.strptime(
+            dt_str, _RFC3339_MICROS).replace(tzinfo=UTC)
+    except ValueError:  # maybe nanosecond resoultion
+        prefix, suffix = dt_str.split('.')
+        fraction, zulu = suffix[:-1], suffix[-1]
```

Comment:
Let me make a stab at requirements clarity.  We have two possible *valid* cases:

- nanosecond precision, `YYYY-mm-ddTHH:MM:SS.nnnnnnnnnZ` (logging API)
- microsecond precision, `YYYY-mm-ddTHH:MM:SS.uuuuuuZ` (all other APIs to date)

The microseconds format can be parsed natively, and fast, using `strptime`.  The second cannot, because Python `datetime` doesn't support nanosecond precision.

There is *no* valid format which does not end with `Z`.  We know none of the other APIs are returning such timestamps, because we don't see the ValueErrors whih would result from the code on `master`.  This PR adds handling for the new, nanosecond-precision case, but doesn't attempt to handle other, hypothetical to-date, error cases.  Do we need to try harder to support non-spec formats?